packages/utils/src/namespace.ts

```diff
@@ -8,7 +8,69 @@ import {
   record,
```

Comment:
I understand that this is great for our side of development, but it's really hard to read if you don't know superstruct -> it'll be hard for our users to understand type definitions for the Provider

Reply:
The types can be inferred by TypeScript fine: 
![image](https://user-images.githubusercontent.com/1561200/192740727-91aa893c-bce0-4df6-b6e0-2100efe06314.png)

Shouldn't that allow users to understand the types or what do you mean?

Reply:
Yep, unlike the auto-generated JSON schema types, the inferred struct types are pretty readable.

<img width="365" alt="image" src="https://user-images.githubusercontent.com/7503723/192744679-88cd3900-9080-4e2d-b054-9666382f9425.png">

That's part of the reason why we chose Superstruct.

Reply:
That's neat! I read projects on Github though, not only on my IDE.

I guess we can leave like that for now